crates/dkg-cli/src/actions.rs

```diff
@@ -100,6 +104,49 @@ pub async fn start(opts: StartOpts) -> Result<()> {
     Ok(())
 }
 
+pub async fn reshare<S, C, R>(opts: ReshareConfig, rng: &mut R) -> Result<()>
+where
+    C: Curve,
+    // We need to bind the Curve's Point and Scalars to the Scheme
+    S: Scheme<Public = <C as Curve>::Point, Private = <C as Curve>::Scalar>,
+    R: RngCore,
+{
+    let provider = Provider::<Http>::try_from(opts.node_url.as_str())?;
+    let client = Arc::new(opts.private_key.parse::<Wallet>()?.connect(provider));
+
+    // we need the previous group and public poly for resharing
+    let previous_group = {
```

Comment:
Does this assume:
1. The previous group save their old private keys 
2. All of the keys are in the qualified set?

Reply:
@nikkolasg and @gakonst clarified that not!